lua/telescope/config.lua

```diff
@@ -109,278 +109,377 @@ local layout_config_description = string.format([[
-      For example:
 
-        ...,
-        ["jj"] = { "<esc>", type = "command" },
-        ["kk"] = { "<cmd>echo \"Hello, World!\"<cr>", type = "command" },)
-        ...,
-    ]],
-  },
+  otherwise, just set the mapping to the function that you want it to
+  be.
 
-  default_mappings = {
-    nil,
-    [[
-    Not recommended to use except for advanced users.
+      ...,
+      ["<C-i>"] = require('telescope.actions').select_default,
+      ...,
 
-    Will allow you to completely remove all of telescope's default maps
-    and use your own.
-    ]],
-  },
+  If the function you want is part of `telescope.actions`, then you can
+  simply give a string.
+    For example, the previous option is equivalent to:
 
-  generic_sorter = { sorters.get_generic_fuzzy_sorter },
-  prefilter_sorter = { sorters.prefilter },
-  file_sorter = { sorters.get_fuzzy_file },
+      ...,
+      ["<C-i>"] = "select_default",
+      ...,
 
-  file_ignore_patterns = { nil },
+  You can also add other mappings using tables with `type = "command"`.
+    For example:
 
-  file_previewer = {
-    function(...)
-      return require("telescope.previewers").vim_buffer_cat.new(...)
-    end,
-  },
-  grep_previewer = {
-    function(...)
-      return require("telescope.previewers").vim_buffer_vimgrep.new(...)
-    end,
-  },
-  qflist_previewer = {
-    function(...)
-      return require("telescope.previewers").vim_buffer_qflist.new(...)
-    end,
-  },
-  buffer_previewer_maker = {
-    function(...)
-      return require("telescope.previewers").buffer_previewer_maker(...)
-    end,
-  },
-}
+      ...,
+      ["jj"] = { "<esc>", type = "command" },
+      ["kk"] = { "<cmd>echo \"Hello, World!\"<cr>", type = "command" },)
+      ...,
+  ]]
+)
+
+append("default_mappings", nil, [[
+  Not recommended to use except for advanced users.
+
+  Will allow you to completely remove all of telescope's default maps
+  and use your own.
+  ]]
+)
+
+append("file_sorter", sorters.get_fzy_sorter, [[
+  A function pointer that specifies the file_sorter. This sorter will
+  be used for find_files, git_files and similar.
+  Hint: If you load a native sorter, you dont need to change this value,
+  the native sorter will override it anyway.
+
+  Default: require("telescope.sorters").get_fzy_sorter]]
+)
+
+append("generic_sorter", sorters.get_fzy_sorter, [[
+  A function pointer to the generic sorter. The sorter that should be
+  used for everything that is not a file.
+  Hint: If you load a native sorter, you dont need to change this value,
+  the native sorter will override it anyway.
+
+  Default: require("telescope.sorters").get_fzy_sorter]]
+)
+
+--TODO(conni2461): Why is this even configurable???
+append("prefilter_sorter", sorters.prefilter, [[
+  This points to a wrapper sorter around the generic_sorter that is able
+  to do prefiltering.
+  Its usually used for lsp_*_symbols and lsp_*_diagnostics
+
+  Default: require("telescope.sorters").prefilter]]
+)
```

Comment:
Still think this should be removed :) 
- To specific
- There is no reason for the user to change it
- We only offer one sorter for prefilter
- If user doesn't want prefiter he can still pass in a different sorter to that lsp_* picker.
- Most likely also not used by any extension because we still miss documentation for sorter.lua (this will change in this pr) + documentation for this config value never existed in the first place

Reply:
ref to myself: https://sourcegraph.com/search?q=context:global+lang:lua+prefilter_sorter&patternType=literal